src/qt/test/optiontests.cpp

```diff
@@ -61,7 +61,10 @@ void OptionTests::migrateSettings()
     QVERIFY(!settings.contains("addrSeparateProxyTor"));
 
     std::ifstream file(gArgs.GetDataDirNet() / "settings.json");
+    std::string default_warning = strprintf("This file is automatically generated and updated by %s. Please do not edit this file while the node "
+                                            "is running, as any changes might be ignored or overwritten.", PACKAGE_NAME);
```

Comment:
I think in general it is safe to edit the file, and main thing worth pointing out is when changes might be ignored or overwritten. It's possible adding more information could be useful, but I don't think there's anything clearly missing. The commit message also seems ok and wouldn't really be an appropriate place to caution users since they are unlikely to see it (in the file itself or in documentation would be better).

Reply:
Since the confusion over the empty settings file error is resolved by the first commit. I agree with ryanofsky; while extending the file warning message sounds plausible, I'm also ok with the current warning.